src/utils.py

```diff
@@ -98,7 +98,11 @@ def buffer_if_necessary(shape):
     Following the advice given here:
     https://github.com/Toblerity/Shapely/issues/344
     """
-    if not shape.is_valid:
-        shape = shape.buffer(0.0)
-    assert shape.is_valid
-    return shape
\ No newline at end of file
+    if shape.is_valid:
+        return shape
+
+    new_shape = shape.buffer(0.0)
+    assert new_shape.is_valid
+    assert np.isclose(new_shape.area, new_shape.area)
+
+    return new_shape
```

Comment:
You really should add newlines to your files. 8-)

Reply:
such a silly rule, but sure, I'll try to remember next time ;) Maybe we can get Hound CI in here to pick these things up for you in future.

Reply:
#10 for the rescue.